src/appleseed-max-impl/logtarget.cpp

```diff
@@ -71,16 +99,29 @@ void LogTarget::write(
         break;
     }
 
-    std::vector<std::string> lines;
-    asf::split(message, "\n", lines);
+    auto is_ui_thread = GetCOREInterface15()->GetMainThreadID() == GetCurrentThreadId();
 
-    for (auto line : lines)
+    if (is_ui_thread)
     {
-        GetCOREInterface()->Log()->LogEntry(
-            type,
-            FALSE,
-            _T("appleseed"),
-            _T("[appleseed] %s"),
-            utf8_to_wide(line).c_str());
+        std::vector<std::string> lines;
+        asf::split(message, "\n", lines);
+        for (auto line : lines)
```

Comment:
This loop is the same in both cases, what about factorizing it into a separate function?